handler/artifact_upload_handler.py

```diff
@@ -45,24 +48,27 @@ def _check_token(self, token):
         # Reduce the risk of timing analysis attacks.
         return hmac.compare_digest(self.hash_expected, hash_result)
 
-    def _index_channel(self):
+    def _index_channel(self, channel):
         """
-        Index the conda channel at `self.write_path` to ensure the channel
-        remains consistent.
+        Index the conda channel to ensure that it remains consistent.
 
         """
-        cmds = [self.conda_exe, 'index', self.write_path]
+        cmds = [self.conda_exe, 'index', channel]
         subprocess.check_call(cmds)
 
     def post(self):
         token = self.get_argument('token')
+        arch = self.get_argument('arch', default='none')
+        arch = '' if arch == 'none' else arch
```

Comment:
How come you are resetting arch already?  Does tornado do the check when you use get_argument()?

Reply:
I've attempted to explain this in https://github.com/SciTools-incubator/artifact-upload-handler/pull/4#discussion_r180974715

Basically, we want `get_argument` to be optional, not required, hence the use of the `default` kwarg.

If the client doesn't send an `arch` argument to the web server in a post, then we don't want the server to complain about it, but just set the `arch` to be an empty string. However, we can't just use an empty string as the default value i.e. `default=''` in the `get_argument` call. The value of `default` much be a non-`False` value.

So the above line will only set `arch` to be the empty string `''` if no value was provided by the client, otherwise `arch` is the value provided by the client.

Reply:
OK, that's all good.  I understood the explanation, I just wanted to check that it was the `get_argument` call that tornado did the check on.  From there it makes perfect sense resetting `'none'` to `' '` on the following line.